### PR TITLE
fix: function parameter and argument separator parsing

### PR DESCRIPTION
**Pull request by feniljain**
- Fix parsing of parameters and argument separator parsing by restricting usage to comma and new line
- Should close https://github.com/SerenityOS/jakt/issues/966
- Would like to hear points on:
https://github.com/feniljain/jakt/blob/f3bc872739fb3a165bd5c30ae0c8d062e6e9af1b/selfhost/parser.jakt#L1465-L1472

I have tried to add appropriate testcases, if you feel something's wrong, do point out!

### CODE REVIEW

File: selfhost/parser.jakt [hunk 1]
```diff
@@ -1458,6 +1461,18 @@ struct Parser {
...
                     current_param_is_mutable = false
                     parameter_complete = false
+
+                    // This would allow syntax like
+                    // function a(
+                    //  a: i64
+                    //
+                    //  b: i64
+                    //)
+                    // Do we want to allow that?
```
Comment (alimpfard) [inline selfhost/parser.jakt:1471]: yes, and please test it too :)
also, pls don't put this sort of code comment in the source kthx

File: samples/functions/parameters_wrong_separation.jakt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/// Expect:
+/// - error: "paramters should be separated by a comma or a newline"
```
Comment (lanmonster) [inline samples/functions/parameters_wrong_separation.jakt:2]: typo. should be "parameters"

File: selfhost/parser.jakt [hunk 3]
```diff
@@ -1502,6 +1517,11 @@ struct Parser {
                     parameter_complete = true
                 }
                 Identifier(name, span) => {
+                    if parameter_complete and not error {
+                        .error("paramters should be separated by a comma or a newline", .current().span())
```
Comment (alimpfard) [inline selfhost/parser.jakt:1521]: Let's make this a "expected <thing>" with that as a hint instead; same for the other commit

Comment (lanmonster) [inline selfhost/parser.jakt:1521]: fix typo here too. also, first letter of the error message should be capital.

File: samples/functions/argument_wrong_separation.jakt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/// Expect:
+/// - error: "arguments should be separated by a comma or a newline"
```
Comment (lanmonster) [inline samples/functions/argument_wrong_separation.jakt:2]: First letter should be capital

File: selfhost/parser.jakt (hunk 1 above)
Comment (feniljain) [reply selfhost/parser.jakt:1471]: Removed them

Have already added tests here:

https://github.com/feniljain/jakt/blob/7e1c263d98388b9650d70be2bf36a2b19efb919e/samples/functions/argument_separation.jakt#L13-L17

and here

https://github.com/feniljain/jakt/blob/7e1c263d98388b9650d70be2bf36a2b19efb919e/samples/functions/parameters_separation_comma_newline.jakt#L15-L21

File: selfhost/parser.jakt (hunk 3 above)
Comment (feniljain) [reply selfhost/parser.jakt]: Screenshot 2022-07-30 at 00 05 14

Currently looks something like this, should I make any changes to this?

**Comment by jntrnr** [comment]
Just to be clear, I'm saying we should not restrict it to only comma and newline. A space should also be okay.

**Comment by feniljain** [comment]
Oh, I misunderstood it the first time. Then I guess these changes are not needed so I will close this PR.

**Pull request closed without merging**